File: mpp/Lab9/csharpPb9/networking/protocol/Response.cs
using networking.DTO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace networking.protocol
{
    [Serializable]
    internal class Response
    {
        [JsonPropertyName("typeCsharp")]
        public ResponseType typeCsharp { get; set; }

        [JsonPropertyName("typeJava")]
        public string typeJava
        {
            get { return typeCsharp.ToString(); }
            set { typeCsharp = (ResponseType)Enum.Parse(typeof(ResponseType), value); }
        }

        [JsonPropertyName("errormessage")]
        public string errormessage { get; set; }
        [JsonPropertyName("user")]
        public UserDTO user { get; set; }
        [JsonPropertyName("rezultat")]
        public RezultatDTO rezultat { get; set; }
        [JsonPropertyName("punctaje")]
        public PunctajParticipantDTO[] punctaje { get; set; }
        [JsonPropertyName("participanti")]
        public PunctajParticipantDTO[] participanti { get; set; }
        public Response() { }
        public override string ToString()
        {
            var punctajeStr = punctaje != null ? string.Join(", ", punctaje.Select(p => p.ToString())) : "null";
            var participantiStr = participanti != null ? string.Join(", ", participanti.Select(p => p.ToString())) : "null";

            return $"Response [type={typeCsharp},typeJava={typeJava}, errormessage={errormessage}, user={user}, rezultat={rezultat}, punctaje=[{punctajeStr}], participanti=[{participantiStr}]]";
        }
    }
}

File: mpp/Lab10/javaPb9/networking/src/main/java/ro/mpp2024/protocol/Response.java
package ro.mpp2024.protocol;

import ro.mpp2024.DTO.PunctajParticipantDTO;
import ro.mpp2024.DTO.RezultatDTO;
import ro.mpp2024.DTO.UserDTO;

import java.io.Serializable;
import java.util.Arrays;

public class Response implements Serializable {
    private ResponseType typeJava;
    private int typeCsharp;
    private String errormessage;
    private UserDTO user;
    private RezultatDTO rezultat;
    private PunctajParticipantDTO[] punctaje;
    private PunctajParticipantDTO[] participanti;

    public Response() {
    }

    public ResponseType getTypeJava() {
        return typeJava;
    }

    public void setTypeJava(ResponseType type) {
        this.typeJava = type;
    }

    public int getTypeCsharp() {
        return CharpTranslateType.toNumber(typeJava);
    }

    public void setTypeCsharp(int typeCsharp) {
        this.typeCsharp = typeCsharp;
    }

    public String getErrormessage() {
        return errormessage;
    }

    public void setErrormessage(String errormessage) {
        this.errormessage = errormessage;
    }

    public UserDTO getUser() {
        return user;
    }

    public void setUser(UserDTO user) {
        this.user = user;
    }

    public PunctajParticipantDTO[] getPunctaje() {
        return punctaje;
    }

    public void setPunctaje(PunctajParticipantDTO[] punctaje) {
        this.punctaje = punctaje;
    }

    public RezultatDTO getRezultat() {
        return rezultat;
    }

    public void setRezultat(RezultatDTO rezultat) {
        this.rezultat = rezultat;
    }

    public PunctajParticipantDTO[] getParticipanti() {
        return participanti;
    }

    public void setParticipanti(PunctajParticipantDTO[] participanti) {
        this.participanti = participanti;
    }

    @Override
    public String toString() {
        return "Response{" +
                "type=" + typeJava +
                ", errormessage='" + errormessage + '\'' +
                ", user=" + user +
                ", rezultat=" + rezultat +
                ", punctaje=" + Arrays.toString(punctaje) +
                '}';
    }
}
